Dockerfile

```diff
@@ -1,18 +1,44 @@
-FROM ubuntu:xenial
+# syntax=docker/dockerfile:experimental
 
-RUN apt-get update && apt-get install --yes nginx
-
-# Set git commit ID
-ARG REVISION_ID
-RUN test -n "${REVISION_ID}"
+# Build stage: Install ruby dependencies
+# ===
+FROM ruby:2.6 AS build-jekyll
+WORKDIR /srv
+ADD . .
+RUN bundle install
+RUN bundle exec jekyll build
 
-# Copy over files
+# Build stage: Install yarn dependencies
+# ===
+FROM node:10-slim AS yarn-dependencies
 WORKDIR /srv
-ADD _site .
+ADD package.json .
+RUN --mount=type=cache,target=/usr/local/share/.cache/yarn yarn install
+
+# Build stage: Run "yarn run build-js"
+# ===
+FROM yarn-dependencies AS build-js
+ADD . .
+RUN yarn run build-js
+
+# Build stage: Run "yarn run build-css"
+# ===
+FROM yarn-dependencies AS build-css
+ADD . .
+RUN yarn run build-css
+
+
+# Import code, build assets and mirror list
```

Comment:
The "circular dependency detected on stage: build-css" error is because there's no `FROM ubuntu:bionic` line to start the main image build.

I'm happy to commit the fix for this.

Reply:
Need to have a look at this later when I got time, because last time I tried it didn't build on my computer.

Reply:
Yeah, I assume the reason I highlighted above would be the reason.

Reply:
This works now I think: https://github.com/canonical-web-and-design/conjure-up.io/pull/61/commits/e638c068c02326cd69327ce4c2f7c9bb2550f394